lighthouse-core/fraggle-rock/api.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Driver = require('./gather/driver.js');
+const Runner = require('../runner.js');
+const Config = require('../config/config.js');
+
+/**
+ * @param {LH.Gatherer.GathererInstance} gatherer
+ * @return {gatherer is LH.Gatherer.FRGathererInstance}
+ */
+function isFRGatherer(gatherer) {
```

Comment:
Are all FR gatherers going to support snapshot mode?

Reply:
No, not all, see [full compat table](https://docs.google.com/spreadsheets/d/1d1oG67eoJWRKMBsYDNMT510_b9J0mPetxsL277k-Jtc/edit#gid=0) for which ones, but the next step after #11759 is the TODO that's here about using gatherer.meta to annotate whether a gatherer supports it or not.

Reply:
The name `isFRGatherer` makes it sound like this will detect FR gatherers, but it's used to detect snapshot support on L60. WDYT of renaming this to something like `isFRSnapshotGatherer`?

Reply:
> The name isFRGatherer makes it sound like this will detect FR gatherers

Excellent, that's exactly what it's supposed to be doing :) purely a type guard on whether a gatherer instance supports the FR gatherer properties.

> but it's used to detect snapshot support on L60

I consider this function not detecting snapshot support at all which is why the TODO is there. Any suggestion for alternative wording there to make it clearer? Snapshot detection support is 2-3 PRs away from being fixed and won't affect any results in the meantime.

Reply:
I think this issue has more to do with me misunderstanding the comment on L59 than the comment's wording. It's fine to leave as is :)